
From: Hans-Frieder Vogt <hfvogt@arcor.de>

I traced the current double-speed issue for 2.6.7-bk2 on x86-64 back to an
ACPI-change in mpparse.c.  The small attached patch solved the issue at
least on my MSI K8T Neo (Athlon 64 3200+) system.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/x86_64/kernel/mpparse.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN arch/x86_64/kernel/mpparse.c~x86_64-double-clock-speed-fix arch/x86_64/kernel/mpparse.c
--- 25/arch/x86_64/kernel/mpparse.c~x86_64-double-clock-speed-fix	2004-06-20 14:39:45.657103608 -0700
+++ 25-akpm/arch/x86_64/kernel/mpparse.c	2004-06-20 14:39:45.661103000 -0700
@@ -861,7 +861,7 @@ void __init mp_config_acpi_legacy_irqs (
 
 		for (idx = 0; idx < mp_irq_entries; idx++)
 			if (mp_irqs[idx].mpc_srcbus == MP_ISA_BUS &&
-				(mp_irqs[idx].mpc_dstapic == ioapic) &&
+				(mp_irqs[idx].mpc_dstapic == intsrc.mpc_dstapic) &&
 				(mp_irqs[idx].mpc_srcbusirq == i ||
 				mp_irqs[idx].mpc_dstirq == i))
 					break;
_
